### PR TITLE
fix(RHINENG-6095): Fix DELETE /hosts endpoint 

### PR DESCRIPTION
**Pull request by jpramos123**
# Overview

This PR is being created to address [RHINENG-6095](https://issues.redhat.com/browse/RHINENG-6095).
This PR reverts an bug that was introduced on #1603 

It reintroduce `stalenessNoDefaultsParam`  to `DELETE /hosts`

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: tests/test_xjoin.py [hunk 1]
```diff
@@ -2228,19 +2228,7 @@ def test_xjoin_search_query_using_hostfilter(
...
-                        {"AND": {"modified_on": mocker.ANY, "spf_host_type": {"eq": "edge"}}},
-                        {"AND": {"modified_on": mocker.ANY, "spf_host_type": {"eq": "edge"}}},
-                        {"AND": {"modified_on": mocker.ANY, "spf_host_type": {"eq": "edge"}}},
-                        {"AND": {"modified_on": mocker.ANY, "spf_host_type": {"eq": None}}},
-                        {"AND": {"modified_on": mocker.ANY, "spf_host_type": {"eq": None}}},
-                        {"AND": {"modified_on": mocker.ANY, "spf_host_type": {"eq": None}}},
-                    )
-                },
-            ),
+            "filter": ({field: {"eq": value}},),
```
Comment (kruai) [inline tests/test_xjoin.py:2231]: Do we not want to test the value of the staleness filter?

Comment (jpramos123) [reply tests/test_xjoin.py:2231]: I think I'm writing a test to specifically test it

Comment (jpramos123) [reply tests/test_xjoin.py:2231]: @kruai Added a test

File: swagger/api.spec.yaml [hunk 2]
```diff
@@ -66,7 +66,7 @@ paths:
         - $ref: '#/components/parameters/updatedEnd'
         - $ref: '#/components/parameters/groupNameListParam'
         - $ref: '#/components/parameters/registered_with'
-        - $ref: '#/components/parameters/stalenessParam'
+        - $ref: '#/components/parameters/stalenessNoDefaultsParam'
```
Comment (thearifismail) [inline swagger/api.spec.yaml:69]: this parameter is not being used any where.  Why not just delete `stalenessNoDefaultsParams` altogether?

**Review by thearifismail — COMMENTED** [review]
One comment else looks good

File: swagger/api.spec.yaml (hunk 2 above)
Comment (jpramos123) [reply swagger/api.spec.yaml:69]: It is being used by `DELETE /hosts`

**Review by FabriciaDinizRH — APPROVED** [review]
LGTM!

**Pull request merged**